crates/uv-extract/src/stream.rs

```diff
@@ -104,9 +104,9 @@ pub async fn unzip<R: tokio::io::AsyncRead + Unpin>(
 /// Unpack the given tar archive into the destination directory.
 ///
 /// This is equivalent to `archive.unpack_in(dst)`, but it also preserves the executable bit.
-async fn untar_in<R: tokio::io::AsyncRead + Unpin, P: AsRef<Path>>(
-    archive: &mut tokio_tar::Archive<R>,
-    dst: P,
+async fn untar_in<'a>(
+    mut archive: tokio_tar::Archive<Box<dyn tokio::io::AsyncRead + Unpin + 'a>>,
```

Comment:
@BurntSushi -- How significant would you expect the penalty to be here for using dynamic dispatch?